File: app/api/track_event.php
<?php
/********************************************
 * Track Event API
 * Records ad views, clicks, and contacts
 ********************************************/
header("Content-Type: application/json");

// Get request data
$input = json_decode(file_get_contents('php://input'), true);
$eventType = $input['event_type'] ?? ''; // 'view', 'click', 'contact'
$adId = $input['ad_id'] ?? '';
$metadata = $input['metadata'] ?? [];

if (empty($eventType) || empty($adId)) {
    http_response_code(400);
    echo json_encode([
        'success' => false,
        'message' => 'Missing required parameters'
    ]);
    exit;
}

$analyticsBase = __DIR__ . "/../companies/analytics/";
if (!is_dir($analyticsBase)) {
    mkdir($analyticsBase, 0775, true);
}

// Create analytics file for this ad if doesn't exist
$analyticsFile = "$analyticsBase/$adId.json";
$analytics = [];

if (file_exists($analyticsFile)) {
    $analytics = json_decode(file_get_contents($analyticsFile), true) ?? [];
}

// Initialize structure if needed
if (!isset($analytics['ad_id'])) {
    $analytics['ad_id'] = $adId;
    $analytics['total_views'] = 0;
    $analytics['total_clicks'] = 0;
    $analytics['total_contacts'] = 0;
    $analytics['events'] = [];
}

// Record event
$event = [
    'type' => $eventType,
    'timestamp' => time(),
    'ip' => $_SERVER['REMOTE_ADDR'] ?? 'unknown',
    'user_agent' => $_SERVER['HTTP_USER_AGENT'] ?? 'unknown',
    'metadata' => $metadata
];

$analytics['events'][] = $event;

// Update counters
switch ($eventType) {
    case 'view':
        $analytics['total_views']++;
        break;
    case 'click':
        $analytics['total_clicks']++;
        break;
    case 'contact':
        $analytics['total_contacts']++;
        $analytics['last_contact'] = time();
        break;
}

$analytics['updated_at'] = time();

// Save analytics
file_put_contents($analyticsFile, json_encode($analytics, JSON_PRETTY_PRINT));

echo json_encode([
    'success' => true,
    'message' => 'Event tracked successfully',
    'event_type' => $eventType,
    'ad_id' => $adId
]);

File: python_system/python_shared/api/track_event.py
"""
track_event.py - Track Event API
Records ad views, clicks, and contacts
Converted from PHP to Python
"""

from fastapi import APIRouter, Request
from fastapi.responses import JSONResponse
from pathlib import Path
import json
import time

router = APIRouter()

ANALYTICS_PATH = Path(__file__).parent.parent / "companies" / "analytics"
ANALYTICS_PATH.mkdir(parents=True, exist_ok=True)


@router.post("/track_event")
async def track_event(request: Request):
    """Track event - matches PHP logic exactly"""
    try:
        data = await request.json()

        event_type = data.get("event_type", "")  # 'view', 'click', 'contact'
        ad_id = data.get("ad_id", "")
        metadata = data.get("metadata", {})

        if not event_type or not ad_id:
            return JSONResponse(
                status_code=400,
                content={"success": False, "message": "Missing required parameters"}
            )

        # Load or create analytics file (matching PHP structure)
        analytics_file = ANALYTICS_PATH / f"{ad_id}.json"
        analytics = {}

        if analytics_file.exists():
            with open(analytics_file, "r") as f:
                analytics = json.load(f)

        # Initialize structure if needed (matching PHP defaults)
        if "ad_id" not in analytics:
            analytics["ad_id"] = ad_id
            analytics["total_views"] = 0
            analytics["total_clicks"] = 0
            analytics["total_contacts"] = 0
            analytics["events"] = []

        # Create event (matching PHP event structure)
        event = {
            "type": event_type,
            "timestamp": int(time.time()),
            "ip": request.client.host if request.client else "unknown",
            "user_agent": request.headers.get("user-agent", "unknown"),
            "metadata": metadata
        }

        analytics["events"].append(event)

        # Update counters (matching PHP switch statement)
        if event_type == "view":
            analytics["total_views"] = analytics.get("total_views", 0) + 1
        elif event_type == "click":
            analytics["total_clicks"] = analytics.get("total_clicks", 0) + 1
        elif event_type == "contact":
            analytics["total_contacts"] = analytics.get("total_contacts", 0) + 1
            analytics["last_contact"] = int(time.time())

        analytics["updated_at"] = int(time.time())

        # Save analytics
        with open(analytics_file, "w") as f:
            json.dump(analytics, f, indent=2)

        return {
            "success": True,
            "message": "Event tracked successfully",
            "event_type": event_type,
            "ad_id": ad_id
        }

    except Exception as e:
        return JSONResponse(
            status_code=500,
            content={"success": False, "error": str(e)}
        )


@router.get("/track_event")
async def track_event_get(ad_id: str, event_type: str, request: Request):
    """Track event via GET request"""
    try:
        if not ad_id or not event_type:
            return JSONResponse(
                status_code=400,
                content={"success": False, "message": "Missing ad_id or event_type"}
            )

        # Load analytics
        analytics_file = ANALYTICS_PATH / f"{ad_id}.json"
        analytics = {
            "ad_id": ad_id,
            "total_views": 0,
            "total_clicks": 0,
            "total_contacts": 0,
            "events": []
        }

        if analytics_file.exists():
            with open(analytics_file, "r") as f:
                analytics = json.load(f)

        # Create event
        event = {
            "type": event_type,
            "timestamp": int(time.time()),
            "ip": request.client.host if request.client else "unknown",
            "user_agent": request.headers.get("user-agent", "unknown")
        }

        analytics["events"].append(event)

        # Update counters
        if event_type == "view":
            analytics["total_views"] = analytics.get("total_views", 0) + 1
        elif event_type == "click":
            analytics["total_clicks"] = analytics.get("total_clicks", 0) + 1
        elif event_type == "contact":
            analytics["total_contacts"] = analytics.get("total_contacts", 0) + 1
            analytics["last_contact"] = int(time.time())

        analytics["updated_at"] = int(time.time())

        # Save
        with open(analytics_file, "w") as f:
            json.dump(analytics, f, indent=2)

        return {
            "success": True,
            "message": "Event tracked successfully",
            "event_type": event_type,
            "ad_id": ad_id
        }

    except Exception as e:
        return JSONResponse(
            status_code=500,
            content={"success": False, "error": str(e)}
        )
